### ISSUE TITLE
Exceptions tracing webClient calls, causes request to hang

### ISSUE THREAD

**Issue by shurdsfield** [opentracing-contrib/java-spring-jaeger#51]

**Comment by geoand** [comment]
Could provide (a lot) more information about the problem please and possible a reproducer project?

**Comment by shurdsfield** [comment]
Steps to create issue

- I add tracing to an existing webflux project 
- that project uses WebClient to call another service
- That response contains invalid headers (or anything that causes an exception)

The result is that there is an exception, but there is no response written and the request hangs without ever terminating

If i remove the tracing dependancy everything works ok and i get a response
I have attached a sample. 

just go to [http://localhost:8080/hello](url)



[spring-cloud-tracing-bug-2.tar.gz](https://github.com/opentracing-contrib/java-spring-jaeger/files/3302325/spring-cloud-tracing-bug-2.tar.gz)

**Comment by geoand** [comment]
I see that you are using a milestone version of Spring Boot. We certainly haven't tested with any such version so it is no surprise that it doesn't work.
Do you see the same behevior if you use the latest release version of Spring Boot?

**Comment by shurdsfield** [comment]
yes, i only tried that to make sure it wasnt a defect that had been fixed but yes changing version makes no difference

**Comment by geoand** [comment]
OK, thanks. I will take a look hopefully soon.

**Comment by geoand** [comment]
@shurdsfield I just tried your reproducer (albeit with Spring Boot `2.1.6.RELEASE`) and the `/hello` endpoint worked without issues.
Can you confirm or is there something missing?

**Comment by shurdsfield** [comment]
i am using jdk : openjdk 11.0.2 
and running
 ./gradlew bootRun

Just tried now and get the exception

**Comment by geoand** [comment]
It worked perfectly for me, here is the last part of the output:

```
2019-06-21 14:07:22.680  INFO 9373 --- [or-http-epoll-2] i.j.internal.reporters.LoggingReporter   : Span reported: b2a4f8c7b7b087ec:b2a4f8c7b7b087ec:0:1 - hello
```

The output of `./gradlew -version` for me is:

```
------------------------------------------------------------
Gradle 5.4.1
------------------------------------------------------------

Build time:   2019-04-26 08:14:42 UTC
Revision:     261d171646b36a6a28d5a19a69676cd098a4c19d

Kotlin:       1.3.21
Groovy:       2.5.4
Ant:          Apache Ant(TM) version 1.9.13 compiled on July 10 2018
JVM:          11.0.2 (Oracle Corporation 11.0.2+9)
OS:           Linux 4.18.0-22-generic amd64
```

**Comment by geoand** [comment]
I also tried with `java -jar ....` after building with `./gradlew build` and same result, the endpoint responsed properly

**Comment by shurdsfield** [comment]
Ok, so it just when using the 2.2 milestone releases i have the issue. I thought id tried with 2.1 release versions  but maybe i was mistaken. something to keep an eye on with the next release

**Comment by geoand** [comment]
Awesome, thanks @shurdsfield for inverstigating.

I will go ahead and close this issue for the time being and we can reopen if there is a problem with the released of version of `2.2.x`

**Comment by shurdsfield** [comment]
Just to add i am blocked since #50 means i cant upgrade to 2.1.6. if i downgrade to io.projectreactor.netty:reactor-netty:0.8.6.RELEASE to fix #50 i  get the exception again

**Comment by geoand** [comment]
So it looks like reactor-netty is doing something weird... I guess we just have to wait and see what happens
